### PR TITLE
Enhancements to groups

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
In no way a MVP feature I just took the opportunity while I was rewriting the create groups page to support the group image and took the easy option of using FA icons.

WIP at this stage bug would like some feedback.

+ added package react-native-check-box
+ added new icon component to allow icons from multiple vector packs
+ Update to the list modal to support icons
+ added icon list to fixtures
+ GQP template updates to support icons
+ GQP template to query for tags with filter

+ Rewrite of the create groups page
   + UI to add icon to new group
   + UI to allow adding tags to new group before creation

+ sorted group response
+ test data updated to have icons
+ server logic to search for tags under org
+ server side logic to create group and pass existing tag
+ added group icons to the data model

screenshot 2018-04-10 12 47 18

screenshot 2018-04-12 21 04 18

screenshot 2018-04-12 21 04 06

### CODE REVIEW

**Comment by sdunster** [comment]
nice PR - looks awesome at a glance! I will review in depth soon.

**Comment by OSPFNeighbour** [comment]
Normally you have way more to say. Have I broken you

**Comment by sdunster** [comment]
nope this was really great! good detail on the changes, self-contained changes.

File: server/src/creators.js [hunk 1]
```diff
@@ -106,10 +106,14 @@ export const getCreators = (models) => {
       }
       return Group.create({
         name,
+        icon: icon || 'group',
         organisationId: organisation.id,
-      }).then(
-        group => group.addUsers(users)
-          .then(() => group));
+      }).then(group => Promise.all([
```
Comment (OSPFNeighbour) [inline server/src/creators.js]: @sdunster this seems to be returning the group before the tags have been added, as in the UI will show now tags on the group until you refetch the data. any ideas?

**Pull request merged**